xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go

```diff
@@ -1138,3 +1143,168 @@ func waitForProducedZeroAddressesError(ctx context.Context, t *testing.T, client
 	}
 	return errors.New("timeout when waiting for RPCs to fail with UNAVAILABLE status and produced zero addresses")
 }
+
+// Test runs a server which listens on multiple ports. The test updates xds resouce
+// cache to contain a single endpoint with multiple addresses. The test intercepts
+// the resolver updates sent to the petiole policy and verifies that the
+// additional endpoint addresses are correctly propagated.
+func (s) TestEDS_EndpointWithMultipleAddresses(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	// Start a backend server which provide that listens to multiple ports to simulate
+	// a backend with multiple addresses.
+	servers, cleanup2 := startTestServiceBackends(t, 1)
+	defer cleanup2()
+	lis1, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Failed to create listener: %v", err)
+	}
+	defer lis1.Close()
+	lis2, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Failed to create listener: %v", err)
+	}
+	defer lis2.Close()
+
+	go servers[0].S.Serve(lis1)
+	go servers[0].S.Serve(lis2)
+
+	addrs, ports := backendAddressesAndPorts(t, servers)
+	additionalPorts := []uint32{
+		testutils.ParsePort(t, lis1.Addr().String()),
+		testutils.ParsePort(t, lis2.Addr().String()),
+	}
```

Comment:
Actually what is happening here is not explicit in the first read because starting the stub server starts one listener and starts serving on it. Then, the test calls `Serve` on two more listeners. So, we have three addresses.

Either make it explicit with a comment, or set up the `Listener` field of the stub server explicitly from the test or some other option that makes it super clear what is happening here.

Thanks.